### PR TITLE
VP-1310: Adapt AuthorizeNet for platform 3.0.0

### PR DESCRIPTION
**Pull request by yecli**
Decided to not create .Core project for the single ModuleConstants and .Data project for the single AuthorizeNetMethod implementation class. 
If better to separate them - please tell.
Also thinking on extracting service for huge API controller `RegisterPayment` method.

### CODE REVIEW

File: src/VirtoCommerce.AuthorizeNet.Web/Controllers/Api/AuthorizeNetController.cs [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+        {
+            _customerOrderService = customerOrderService;
+            _storeService = storeService;
+            _paymentMethodsSearchService = paymentMethodsSearchService;
+        }
+
+        [HttpPost]
+        [Route("registerpayment/{orderId}")]
+        [AllowAnonymous]
+        public async Task<ActionResult> RegisterPayment(string orderId)
```
Comment (akak1977) [inline src/VirtoCommerce.AuthorizeNet.Web/Controllers/Api/AuthorizeNetController.cs:26]: Please mark action with swagger-related attributes.
```csharp
        [ProducesResponseType(typeof(void), StatusCodes.Status204NoContent)]
        [ProducesResponseType(typeof(string), StatusCodes.Status200OK)]
```

**Review by akak1977 — APPROVED** [review]

**Review by akak1977 — APPROVED** [review]

File: src/VirtoCommerce.AuthorizeNet.Web/module.manifest [hunk 2]
```diff
@@ -0,0 +1,30 @@
+<?xml version="1.0" encoding="utf-8" ?>
+<module>
+    <id>VirtoCommerce.AuthorizeNet</id>
+    <version>3.0.0</version>
+	  <version-tag>rc.3</version-tag>
+    <platformVersion>3.0.0</platformVersion>
+    <dependencies>
+      <dependency id="VirtoCommerce.Store" version="3.0.0" />
+			<dependency id="VirtoCommerce.Orders" version="3.0.0-rc.3-001" />
```
Comment (tatarincev) [inline src/VirtoCommerce.AuthorizeNet.Web/module.manifest]: Use version="3.0.0" for all dependencies

File: src/VirtoCommerce.AuthorizeNet.Web/Services/AuthorizeNetRegisterPaymentService.cs [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+using VirtoCommerce.OrdersModule.Core.Services;
+using VirtoCommerce.PaymentModule.Core.Model.Search;
+using VirtoCommerce.PaymentModule.Core.Services;
+using VirtoCommerce.PaymentModule.Model.Requests;
+using VirtoCommerce.Platform.Core.Common;
+using VirtoCommerce.StoreModule.Core.Services;
+
+namespace VirtoCommerce.AuthorizeNet.Web.Services
+{
+    public class AuthorizeNetRegisterPaymentService : IAuthorizeNetRegisterPaymentService
```
Comment (tatarincev) [inline src/VirtoCommerce.AuthorizeNet.Web/Services/AuthorizeNetRegisterPaymentService.cs:17]: Why do we need this separated abstraction, is it impossible to move into PaymentMethod?

File: src/VirtoCommerce.AuthorizeNet.Web/module.manifest (hunk 2 above)
Comment (yecli) [reply src/VirtoCommerce.AuthorizeNet.Web/module.manifest]: Done

File: src/VirtoCommerce.AuthorizeNet.Web/Services/AuthorizeNetRegisterPaymentService.cs (hunk 3 above)
Comment (yecli) [reply src/VirtoCommerce.AuthorizeNet.Web/Services/AuthorizeNetRegisterPaymentService.cs:17]: It defines payment result handling workflow. 
It was extracted from the old controller to facilitate controller-contained logic:
https://github.com/VirtoCommerce/vc-module-Authorize.Net/pull/5/files/6f5afbab40be3ff01eff211e68743dc1d900e594#diff-9a6d5d29037c4d28b6805d97050723afL33

Not sure we need to move it to PaymentMethod, as it has the methods for specific actions (Process, PostProcess, Refund etc.). New abstraction is like a manager calling different PaymentMethod's operations.
Almost the same code excluding AuthorizeNet specific and return type exists in Orders module:
https://github.com/VirtoCommerce/vc-module-order/blob/release/3.0.0/src/VirtoCommerce.OrdersModule.Web/Controllers/Api/OrderModuleController.cs#L419

So I think we could have some abstraction for this logic to use it in all places.

Comment (tatarincev) [reply src/VirtoCommerce.AuthorizeNet.Web/Services/AuthorizeNetRegisterPaymentService.cs:17]: Ok, then. Anyway, we will change this logic soon into an absolutely different manner.

**Pull request merged**